src/databricks/sql/client.py

```diff
@@ -1,7 +1,7 @@
 import decimal
 import json
 import os
-from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union, TYPE_CHECKING
```

Comment:
Why are we doing this pattern?

Reply:
[This is strictly for type checking](https://stackoverflow.com/questions/61545580/how-does-mypy-use-typing-type-checking-to-resolve-the-circular-import-annotation). It allows us to perform circular imports of types across files without needing to reorganise the code for the sole purpose of satisfying mypy.

Reply:
I would suggest that this is generally a code smell.  I don't think its worth blocking on, since you are only relying on it to provide types for type checking, but whenever you hit a situation like this, it's worth considering if there is another organization scheme that brings related types closer together, or structuring the code into layers with a single dependency direction.